packages/react-positioning/src/types.ts

```diff
@@ -0,0 +1,190 @@
+import * as PopperJs from '@popperjs/core';
+import * as React from 'react';
+
+// Temporary typings for modifiers
+
+type ModifierProps<Name, Options> = {
+  enabled?: boolean;
+  name: Name;
+  options?: Options;
+};
+
+export type ArrowModifier = ModifierProps<
+  'arrow',
+  Partial<{
+    element: HTMLElement | string;
+    padding: PopperJs.Padding;
+  }>
+>;
+
+export type FlipModifier = ModifierProps<
+  'flip',
+  Partial<{
+    fallbackPlacements: PopperJs.Placement[];
+    padding: PopperJs.Padding;
+    boundary: PopperJs.Boundary;
+    rootBoundary: PopperJs.RootBoundary;
+    altBoundary: boolean;
+    flipVariations: boolean;
+  }>
+>;
+
+export type OffsetFunctionParam = {
+  popper: PopperJs.Rect;
+  reference: PopperJs.Rect;
+  placement: PopperJs.Placement;
+};
+
+export type OffsetFunction = (param: OffsetFunctionParam) => [number | null | undefined, number | null | undefined];
+
+export type Offset = OffsetFunction | [number | null | undefined, number | null | undefined];
+
+export type OffsetModifier = ModifierProps<
+  'offset',
+  Partial<{
+    offset: Offset;
+  }>
+>;
+
+export type PreventOverflowModifier = ModifierProps<
+  'preventOverflow',
+  Partial<{
+    mainAxis: boolean;
+    altAxis: boolean;
+    boundary: PopperJs.Boundary;
+    rootBoundary: PopperJs.RootBoundary;
+    altBoundary: boolean;
+    /**
+     * Allows the popper to overflow from its boundaries to keep it near its
+     * reference element
+     */
+    tether: boolean;
+    tetherOffset: (arg0: { popper: PopperJs.Rect; reference: PopperJs.Rect; placement: PopperJs.Placement }) => number;
+    padding: PopperJs.Padding;
+  }>
+>;
+
+export type PopperModifiers = (ArrowModifier | FlipModifier | OffsetModifier | PreventOverflowModifier)[];
+export type PopperModifiersFn = (
+  target: HTMLElement | PopperJs.VirtualElement,
+  container: HTMLElement,
+  arrow: HTMLElement | null,
+) => PopperModifiers;
```

Comment:
```suggestion
```

As we have agreed to not expose `modifiers` yet, we can remove these types. I think that temporary types (`(ArrowModifier, `FlipModifier`, `OffsetModifier`, `PreventOverflowModifier`) can be also deleted in this case.